pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/SimpleLoadManagerImpl.java

```diff
@@ -236,19 +228,12 @@ public void initialize(final PulsarService pulsar) {
         lastLoadReport.setPersistentTopicsEnabled(pulsar.getConfiguration().isEnablePersistentTopics());
         lastLoadReport.setNonPersistentTopicsEnabled(pulsar.getConfiguration().isEnableNonPersistentTopics());
 
-        loadReportCacheZk = new ZooKeeperDataCache<LoadReport>(pulsar.getLocalZkCache()) {
-            @Override
-            public LoadReport deserialize(String key, byte[] content) throws Exception {
-                return ObjectMapperFactory.getThreadLocal().readValue(content, LoadReport.class);
-            }
-        };
-        loadReportCacheZk.registerListener(this);
-        this.dynamicConfigurationCache = new ZooKeeperDataCache<Map<String, String>>(pulsar.getLocalZkCache()) {
-            @Override
-            public Map<String, String> deserialize(String key, byte[] content) throws Exception {
-                return ObjectMapperFactory.getThreadLocal().readValue(content, HashMap.class);
-            }
-        };
+        loadReports = pulsar.getCoordinationService().getLockManager(LoadReport.class);
+        pulsar.getLocalMetadataStore().registerListener(this);
+        this.dynamicConfigurationCache = pulsar.getLocalMetadataStore().getMetadataCache(
+                new TypeReference<Map<String, String>>() {
```

Comment:
for the future:
here we are subclassing TypeReference with an anonymous class. can we provide a better API for this case ?